core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -376,8 +376,6 @@ private[spark] class MapOutputTrackerMaster(conf: SparkConf)
    * @param numReducers total number of reducers in the shuffle
    * @param fractionThreshold fraction of total map output size that a location must have
    *                          for it to be considered large.
-   *
-   * This method is not thread-safe.
```

Comment:
I don't see any reason to believe it's not thread-safe; it's kind of neither explicitly thread-safe nor explicitly unsafe IMO. It's possible that individual `status` entries might change their size from underneath this function, but I don't see any problem with that. I am not necessarily the best person to ask, though.